### ISSUE TITLE
QTable has problems reading in ipac format when a file has both units and strings

### ISSUE THREAD

**Issue by lauralwatkins** [astropy/astropy#4867]
I'm having a problem where I cannot read in an ipac-formatted file as a QTable when at least one column contains strings and at least one column has a unit.

For example...

Let's create a QTable with a column of strings and a column with a unit:
```python
foo = table.QTable()
foo["name"] = ["thing1", "thing2", "thing3"]
foo["mass"] = np.random.rand(3)*u.Msun
foo.write("myfile.dat", format="ascii.ipac")
```

This creates a file that looks like this:
```
|  name|          mass|
|  char|        double|
|      |       solMass|
|  null|          null|
 thing1 0.667003071472 
 thing2 0.474236083016 
 thing3 0.778330921581 
```

Now let's read the file in again:
```python
bar = table.QTable.read("myfile.dat", format="ascii")
```

And enjoy the error message:
```
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-15-9a2bb1608e08> in <module>()
----> 1 bar = table.QTable.read("myfile.dat", format="ascii")

~/Library/Python/2.7/lib/python/site-packages/astropy/table/table.pyc in read(cls, *args, **kwargs)
   2259         passed through to the underlying data reader (e.g. `~astropy.io.ascii.read`).
   2260         """
-> 2261         return io_registry.read(cls, *args, **kwargs)
   2262 
   2263     def write(self, *args, **kwargs):

~/Library/Python/2.7/lib/python/site-packages/astropy/io/registry.pyc in read(cls, *args, **kwargs)
    353                 except:
    354                     raise TypeError('could not convert reader output to {0} class'
--> 355                                     .format(cls.__name__))
    356             else:
    357                 raise TypeError("reader should return a {0} instance".format(cls.__name__))

TypeError: could not convert reader output to QTable class
```

**Comment by eteq** [comment]
Taking a quick look at this, I think I see the problem.  the IPAC format produces a header row for units, which is good... But it uses a blank space as a placeholder for "no unit".  So Table parses th units for the 'name' column in this example as ``Unit('')``, which is the *dimensionless* unit, rather than "None", which is what Quantity expects for "no unit".  Then when it converts to QTable it fails because you can't apply a unit (even dimensionless) to a string array.

I *think* this fix is to just make it so that an empty string in the unit part of the  header yields `None` for the unit instead of ``Unit('')``.  Do you agree, @taldcroft ?  If so, @lauralwatkins has said she's happy to create a PR to do that.

**Comment by taldcroft** [comment]
@eteq @lauralwatkins - Yup, that does the trick.  It looks like the fix is really simple since `''` evaluates to False and the boolean chaining will end up evaluating to `None`.
```
@@ -193,7 +193,7 @@ class IpacHeader(fixedwidth.FixedWidthHeader):
                 col.raw_type = header_vals[1][i].strip(' -')
                 col.type = self.get_col_type(col)
             if len(header_vals) > 2:
-                col.unit = header_vals[2][i].strip()  # Can't strip dashes here
+                col.unit = header_vals[2][i].strip() or None  # Can't strip dashes here
             if len(header_vals) > 3:
                 # The IPAC null value corresponds to the io.ascii bad_value.
                 # In this case there isn't a fill_value defined, so just put
```
@lauralwatkins - a quick test just like showed, but using a StringIO buffer instead of a real file, should do the trick!  You can look in `astropy/io/ascii/tests/test_ipac_definitions.py` for examples and just add a new test in that file.  Code freeze is soon (though bug fixes should get more leeway).  :smile:

**Comment by taldcroft** [comment]
Closed by #4947.
